tests/php/sync/test_class.jetpack-sync-callables.php

```diff
@@ -1151,6 +1151,49 @@ public function test_get_main_network_site_wpcom_id_multisite() {
 		restore_current_blog();
 	}
 
+	/**
+	 * Verify get_check_sum is consistent for differently ordered arrays.
+	 */
+	public function test_sync_does_not_send_updates_if_array_order_is_only_change() {
+		$plugins = Functions::get_plugins();
+
+		// Let's see if the original values get synced.
+		$this->sender->do_sync();
+		$plugins_synced = $this->server_replica_storage->get_callable( 'get_plugins' );
+
+		$this->assertEquals( $plugins, $plugins_synced );
+
+		add_filter( 'all_plugins', array( $this, 'randomize_array_keys' ), 100, 1 );
+		do_action( 'jetpack_sync_unlock_sync_callable' );
+		$this->server_event_storage->reset();
+		$this->sender->do_sync();
+
+		$event = $this->server_event_storage->get_most_recent_event( 'jetpack_sync_callable' );
+		$this->assertFalse( $event );
+	}
+
+	/**
+	 * Randomize the get_plugins array keys.
+	 *
+	 * @param array $plugins array of plugins.
+	 *
+	 * @return array
+	 */
+	public function randomize_array_keys( $plugins ) {
+		// First plugin in array.
+		$plugin_key = array_keys( $plugins )[0];
+
+		// sort the plugin's array entries.
+		$keys = array_keys( $plugins[ $plugin_key ] );
+		shuffle( $keys );
```

Comment:
It's possible that this doesn't actually change the order of the keys, particularly if `$keys` has few entries.

In general, randomness in tests is bad as it makes test runs non-reproducible.

Reply:
Excellent point, updated to use array_reverse and renamed filter function to align to no longer being random.

Thanks